src/components/forms/NumberField/NumberField.tsx

```diff
@@ -0,0 +1,185 @@
+import {FormField, Textbox} from '@utrecht/component-library-react';
+import DOMPurify from 'dompurify';
+import {useField, useFormikContext} from 'formik';
+import {useId} from 'react';
+import {useIntl} from 'react-intl';
+import {NumericFormat} from 'react-number-format';
+
+import HelpText from '@/components/forms/HelpText';
+import Label from '@/components/forms/Label';
+import Tooltip from '@/components/forms/Tooltip';
+import ValidationErrors from '@/components/forms/ValidationErrors';
+
+import './_input-container.scss';
+
+export interface NumberFieldProps {
+  /**
+   * The name of the form field/input, used to set/track the field value in the form state.
+   */
+  name: string;
+  /**
+   * The (accessible) label for the field - anything that can be rendered.
+   *
+   * You must always provide a label to ensure the field is accessible to users of
+   * assistive technologies.
+   */
+  label: React.ReactNode;
+  /**
+   * Required fields get additional markup/styling to indicate this validation requirement.
+   */
+  isRequired?: boolean;
+  /**
+   * Disabled fields get marked as such in an accessible manner. Note that Form.io uses
+   * 'disabled' in the component definition, but we would like to have the behaviour of
+   * a read-only field so we use the readOnly prop internally.
+   *
+   */
+  isDisabled?: boolean;
```

Comment:
already here the prop should be renamed to `isReadonly` - at the Formio layer (registry) we can do: `isReadonly={component.disabled}` where the translate Formio's oopsie to proper semantics